tools/genconversion/conversion.go

```diff
@@ -1,90 +1,56 @@
+// conversion-gen is a tool for auto-generating Conversion functions.
+//
+// Structs in the input directories with the below line in their comments
+// will be ignored during generation.
+// // +genconversion=false
 package main
 
 import (
-	"fmt"
-	"io"
-	"log"
-	"os"
-	"runtime"
-	"sort"
-	"strings"
-
-	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/api/unversioned"
-	pkg_runtime "k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/cmd/libs/go2idl/args"
+	"k8s.io/kubernetes/cmd/libs/go2idl/conversion-gen/generators"
 
 	"github.com/golang/glog"
-	flag "github.com/spf13/pflag"
-
-	_ "github.com/openshift/origin/pkg/api"
-	_ "github.com/openshift/origin/pkg/api/v1"
-	_ "github.com/openshift/origin/pkg/api/v1beta3"
-
-	// install all APIs
-	_ "github.com/openshift/origin/pkg/api/install"
-	_ "k8s.io/kubernetes/pkg/api/install"
-	_ "k8s.io/kubernetes/pkg/apis/extensions/install"
-)
-
-var (
-	functionDest = flag.StringP("funcDest", "f", "-", "Output for conversion functions; '-' means stdout")
-	group        = flag.StringP("group", "g", "", "Group for conversion.")
-	version      = flag.StringP("version", "v", "v1beta3", "Version for conversion.")
 )
 
 func main() {
-	runtime.GOMAXPROCS(runtime.NumCPU())
-	flag.Parse()
-	log.SetOutput(os.Stderr)
-
-	var funcOut io.Writer
-	if *functionDest == "-" {
-		funcOut = os.Stdout
-	} else {
-		file, err := os.Create(*functionDest)
-		if err != nil {
-			glog.Fatalf("Couldn't open %v: %v", *functionDest, err)
-		}
-		defer file.Close()
-		funcOut = file
+	arguments := args.Default()
+
+	// Override defaults. These are Kubernetes specific input locations.
+	arguments.InputDirs = []string{
```

Comment:
We only depend on resources in that group today - the list of kube
resources here is the ones that should be considered for input
dependencies, not necessarily for general regeneration.

On Wed, May 4, 2016 at 11:41 AM, Maciej Szulik <notifications@github.com>
wrote:

> In tools/genconversion/conversion.go
> <https://github.com/openshift/origin/pull/8703#discussion_r62062685>:
>
> > -	log.SetOutput(os.Stderr)
> > -
> > -	var funcOut io.Writer
> > -	if *functionDest == "-" {
> > -		funcOut = os.Stdout
> > -	} else {
> > -		file, err := os.Create(*functionDest)
> > -		if err != nil {
> > -			glog.Fatalf("Couldn't open %v: %v", *functionDest, err)
> > -		}
> > -		defer file.Close()
> > -		funcOut = file
> > +	arguments := args.Default()
> > +
> > +	// Override defaults. These are Kubernetes specific input locations.
> > +	arguments.InputDirs = []string{
>
> Do we care only about k8s v1, or other groups that are there should be
> included as well?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8703/files/2917084e22de3e20a191c6345080283e742496b4..3dc8ce14edd91c395c9d6838be7093ce8bac6953#r62062685>
>